lib/perl/Genome/Model/Tools/CgHub/Metadata.pm

```diff
@@ -13,175 +13,143 @@ class Genome::Model::Tools::CgHub::Metadata {
-    my $self = shift;
-
-    my $reference_shortname = $self->get_attribute_value('refassem_short_name');
-    return $reference_shortname if $reference_shortname;
-
-    $reference_shortname = $self->metadata->{Result}->{analysis_xml}->{ANALYSIS_SET}->{ANALYSIS}->{ANALYSIS_TYPE}->{REFERENCE_ALIGNMENT}->{ASSEMBLY}->{STANDARD}->{shortname};
-    return $reference_shortname if $reference_shortname;
+    my $metadata = XMLin($xml);
+    die $self->error_message('Failed to load metadata XML!') if not $metadata;
+    $self->_add_metadata($metadata);
 
-    return;
+    return 1;
 }
 
-sub reference_assembly_version {
-    my $self = shift;
-
-    my $reference_assembly_shortname = $self->reference_assembly_shortname;
-    return if not $reference_assembly_shortname;
-
+sub _add_metadata {
+    my ($self, $incoming_metadata) = Params::Validate::validate_pos(@_, {type => OBJECT}, {type => HASHREF, called => 'metadata',});
 
-    if ( $reference_assembly_shortname =~ /hg(\d\d)/i ) {
-        return $1 + 18;
+    my $results;
+    if ( not $incoming_metadata->{Hits} or $incoming_metadata->{Hits} == 0) {
+        return 1;
     }
-    elsif ( $reference_assembly_shortname =~ /ncbi(\d\d)/i ) {
-        return $1;
+    elsif ( $incoming_metadata->{Hits} == 1) {
+        $results = [ $incoming_metadata->{Result} ];
+    }
+    else {
+        $results = [ values %{$incoming_metadata->{Result}} ];
     }
 
-    return;
+    my $metadata = $self->metadata;
+    for my $result ( @$results ) {
+        for my $id_by (qw/ analysis_id legacy_sample_id /) {
+            $metadata->{ $result->{$id_by} } = $result;
+        }
+    }
+    $self->metadata($metadata);
+
+    return 1;
 }
 
-sub target_region {
-    # This will need to be updated
-    my $self = shift;
+sub get_attribute_value {
+    my ($self, $lookup_id, $name) = Params::Validate::validate_pos(
+        @_, {type => HASHREF}, {type => SCALAR}, {type => SCALAR},
+    );
 
-    die $self->error_message('No metadata set to get target region!') if not $self->metadata;
-
-    my $target_region;
-    my $library_strategy = $self->get_attribute_value('library_strategy');
-    die $self->error_message('No library strategy in metadata to resolve target region!') if not $library_strategy;
-    if ( $library_strategy eq 'WGS' or $library_strategy eq 'RNA-Seq' ) {
-        $target_region = 'none';
-    } elsif($library_strategy eq 'WXS') { #exome
-        my $reference_assembly_version = $self->reference_assembly_version;
-        $target_region = ( $reference_assembly_version eq '37' )
-        ? 'agilent_sureselect_exome_version_2_broad_refseq_cds_only_hs37'
-        : 'agilent sureselect exome version 2 broad refseq cds only';
-    }
-    else {
-        # unknown library strategy
-        $target_region = 'none';
-    }
+    my $result = $self->metadata->{$lookup_id};
+    return if not $result;
+    my $value = $result->{$name};
 
-    return $target_region;
+    return if not defined $value or $value eq '';
```

Comment:
Yeah - There is some code leftover as I updating having the metadata hold multiple analysis ids.